src/modules/rewards/services/arb-rebate-service.ts

```diff
@@ -36,12 +41,110 @@ export class ArbRebateService {
   constructor(
     @InjectRepository(Deposit) readonly depositRepository: Repository<Deposit>,
     @InjectRepository(ArbReward) readonly arbRewardRepository: Repository<ArbReward>,
+    @Inject(CACHE_MANAGER) private cacheManager: Cache,
     private marketPriceService: MarketPriceService,
     private ethProvidersService: EthProvidersService,
     private appConfig: AppConfig,
     private dataSource: DataSource,
   ) {}
 
+  public async getEarnedRewards(userAddress: string) {
+    userAddress = assertValidAddress(userAddress);
+
+    const data = await this.cacheManager.get(getArbEarnedRewardsCacheKey(userAddress));
+    if (data) return data;
+
+    const baseQuery = this.buildBaseQuery(this.arbRewardRepository.createQueryBuilder("r"), userAddress);
+    const { arbRewards } = await baseQuery
+      .select("SUM(CAST(r.amount as DECIMAL))", "arbRewards")
```

Comment:
I think here we should add a WHERE condition to filter only unclaimed rewards

Reply:
Yes, I agree, I thought we were using both unclaimed and claimed in the earned referral rewards service but we are not. I've updated both arb and op query.